File: src/server/Role.py
class Role():
    """The implementation of a exemplary role class.
    """

    def __init__(self):
        self._id = 0
        self._name = ""   

    def get_id(self):
        """Reads out the id."""
        return self._id

    def set_id(self, val):
        """Sets the id."""
        self._id = val
    
    def get_name(self):
        """Reads out the name."""
        return self._name

    def set_name(self, txt):
        """Sets the name."""
        self._name = txt

    @staticmethod
    def from_dict(dicti=dict()):

        """"Converts a Python dict() into a Role()."""
        role = Role()
        role.set_id(dicti["id"])
        role.set_name(dicti["name"])
        return role

File: frontend/src/api/Role.js
export default class Role {

    /// Represents a Role
    
        constructor(aId, aName){
            this.id=aId;
            this.name=aName;
        }
    
        //Sets the id
        setID(aID){
          this.id = aID;
        }
    
        //Reads out the id
        getID(){
            return this.id;
        }

        //Sets the name
        setName(aName){
            this.name = aName;
          }
      
        //Reads out the name
        getName(){
            return this.name;
        }
    
        //Returns an Array of Roles from a given JSON structure.
    
        static fromJSON(roles) {
            let result = [];
    
            if (Array.isArray(roles)) {
                roles.forEach((s) => {
                    Object.setPrototypeOf(s, State.prototype);
                    result.push(s);
                })
            }
            // it's a single object and not an array
            else {
                let s = roles;
                Object.setPrototypeOf(s, State.prototype);
                result.push(s);
            }
    
            return result;
        }
    }